assets/js/atomic/blocks/product/add-to-cart/shared/add-to-cart-button.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * External dependencies
+ */
+import { __, _n, sprintf } from '@wordpress/i18n';
+import Button from '@woocommerce/base-components/button';
+import { Icon, done as doneIcon } from '@woocommerce/icons';
+import { useState } from '@wordpress/element';
+import { useAddToCartFormContext } from '@woocommerce/base-context';
+
+/**
+ * Add to Cart Form Qty + Button Block Component.
+ */
+const AddToCartButton = () => {
+	const {
+		showFormElements,
+		product,
+		quantityInCart,
+		formDisabled,
+		formSubmitting,
+		onSubmit,
+	} = useAddToCartFormContext();
+
+	const {
+		is_purchasable: isPurchasable = true,
+		has_options: hasOptions,
+		add_to_cart: addToCartButtonData = {
+			url: '',
+			text: '',
+		},
+	} = product;
+
+	if ( ( showFormElements || ! hasOptions ) && isPurchasable ) {
+		return (
+			<ButtonComponent
+				className="wc-block-components-product-add-to-cart-button"
+				quantityInCart={ quantityInCart }
+				disabled={ formDisabled }
+				loading={ formSubmitting }
+				onClick={ onSubmit }
+			/>
+		);
+	}
+
+	return (
+		<LinkComponent
+			className="wc-block-components-product-add-to-cart-button"
+			href={ addToCartButtonData.url }
+			text={
+				addToCartButtonData.text ||
+				__( 'View Product', 'woo-gutenberg-products-block' )
+			}
+		/>
+	);
+};
+
+/**
+ * Button for non-purchasable products.
+ */
+const LinkComponent = ( { className, href, text } ) => {
+	return (
+		<Button className={ className } href={ href } rel="nofollow">
+			{ text }
+		</Button>
+	);
+};
+
+/**
+ * Button for purchasable products.
+ */
+const ButtonComponent = ( {
+	className,
+	quantityInCart,
+	loading,
+	disabled,
+	onClick,
+} ) => {
+	const [ wasClicked, setWasClicked ] = useState( false );
```

Comment:
This is show we only show qty in cart after using the add to cart button.

Reply:
Curious - what's the problem with showing the amount in cart on page refresh? I guess that's a design question, but feels appropriate to me. If you've added the product to cart, the button consistently shows you this.

Reply:
I wanted to make sure it was shown as an "Add to Cart" button on first load like the mocks, rather than a button which says "x in cart". I think design will need another pass over all the blocks when we have them all built :)